app/src/main/java/com/github/se/orator/ui/friends/FriendsScreen.kt

```diff
@@ -362,6 +378,39 @@ fun ViewFriendsScreen(
                     }
                   }
 
+              // Display the BattlePopup if a battle is selected
+              selectedBattle?.let { battle ->
+                // Extract the InterviewContext from the SpeechBattle
+                val battleContext = battle.context
+                // Get the challenger's name
+                val challengerName = userProfileViewModel.getName(battle.challenger)
+                BattlePopup(
+                    challengerName = challengerName,
+                    interviewContext = battleContext,
+                    onAccept = {
+
+                      // Handle battle acceptance
+                      battleViewModel?.acceptBattle(battle.battleId)
+                      selectedBattle = null
+
+                      // Show a toast
+                      Toast.makeText(localContext, "Battle accepted!", Toast.LENGTH_SHORT).show()
```

Comment:
What if there is an error when performing firestore operations ? It looks like the toast is still displayed regardless.

Reply:
You're totally right, and I’ll make sure to fix this. Do you have any ideas or suggestions on how I can improve it?

Reply:
The way you did it is good ! (creating the toast in `acceptBattle`)

Reply:
Anyways I removed it because I think that it was causing a bug when accepting the battles!